api/legacy.go

```diff
@@ -9,11 +9,13 @@ import (
 	"github.com/ONSdigital/log.go/v2/log"
 )
 
-type LegacyResponse struct {
-	URL string `json:"url,omitempty"`
+// ClientError is an interface that can be used to retrieve the status code if a client has errored
+type ClientError interface {
```

Comment:
Not sure about this - what errors are shaping? - I cannot see any 'service' errors that would implement this (this is related to the setStatusCode() below

Reply:
Thanks Patrick. You are right, this should not be here. Removed